torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1196,6 +1196,14 @@ struct to_ir {
               return emitHasAttr(apply.inputs()[0], apply.inputs()[1]);
             }
           }
+          auto sv = emitSugaredExpr(apply.callee(), 1);
+          auto loc = apply.callee().range();
+          if (auto special_form = dynamic_cast<SpecialFormValue*>(sv.get())) {
+            if (special_form->form() == prim::isinstance) {
+              checkApplyNumInputs(apply, 2);
+              return emitIsInstance(apply.inputs()[0], apply.inputs()[1]);
+            }
+          }
```

Comment:
This is where the type refinement is happening. I'm not sure if what I did to extract the identify of the torch.jit.isinstance case with lines 1199-1202 is the cleanest way that it could be done